packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -0,0 +1,346 @@
+/**
+ * Completion provider for a code session
+ */
+import { PureComponent } from 'react';
+import * as monaco from 'monaco-editor';
+import Log from '@deephaven/log';
+import { IdeSession } from '@deephaven/jsapi-shim';
+
+const log = Log.module('MonacoCompletionProvider');
+
+interface MonacoProviderProps {
+  model: monaco.editor.ITextModel;
+  session: IdeSession;
+  language: string;
+}
+
+/**
+ * Registers a completion provider with monaco for the language and session provided.
+ */
+class MonacoProviders extends PureComponent<
+  MonacoProviderProps,
+  Record<string, never>
+> {
+  /**
+   * Converts LSP CompletionItemKind to Monaco CompletionItemKind
+   * @param kind The LSP kind
+   * @returns Monaco kind
+   */
+  static lspToMonacoKind(kind: number): number {
+    const monacoKinds = monaco.languages.CompletionItemKind;
+    switch (kind) {
+      case 1:
+        return monacoKinds.Text;
+      case 2:
+        return monacoKinds.Method;
+      case 3:
+        return monacoKinds.Function;
+      case 4:
+        return monacoKinds.Constructor;
+      case 5:
+        return monacoKinds.Field;
+      case 6:
+        return monacoKinds.Variable;
+      case 7:
+        return monacoKinds.Class;
+      case 8:
+        return monacoKinds.Interface;
+      case 9:
+        return monacoKinds.Module;
+      case 10:
+        return monacoKinds.Property;
+      case 11:
+        return monacoKinds.Unit;
+      case 12:
+        return monacoKinds.Value;
+      case 13:
+        return monacoKinds.Enum;
+      case 14:
+        return monacoKinds.Keyword;
+      case 15:
+        return monacoKinds.Snippet;
+      case 16:
+        return monacoKinds.Color;
+      case 17:
+        return monacoKinds.File;
+      case 18:
+        return monacoKinds.Reference;
+      case 19:
+        return monacoKinds.Folder;
+      case 20:
+        return monacoKinds.EnumMember;
+      case 21:
+        return monacoKinds.Constant;
+      case 22:
+        return monacoKinds.Struct;
+      case 23:
+        return monacoKinds.Event;
+      case 24:
+        return monacoKinds.Operator;
+      case 25:
+        return monacoKinds.TypeParameter;
+      default:
+        return monacoKinds.Variable;
```

Comment:
Shouldn't we throw here?

Reply:
I opted not to since kind is optional from the LSP spec. We don't actually provide a kind from groovy completion items. Monaco will show something and I believe it was Variable by default, but I'd have to double check

I'll adjust the param to be `number | undefined`